src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
```

Comment:
For the `updateFragment` function you suggested, what functionality will it have? I don't know how to divide the logic separately.